spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkCopyOnWriteScan.java

```diff
@@ -102,6 +103,14 @@ public NamedReference[] filterAttributes() {
 
   @Override
   public void filter(Filter[] filters) {
+    Preconditions.checkState(
```

Comment:
I initially thought about not doing runtime filtering if the table has been refreshed to avoid failing the query. However, we change such a critical place. I think it is better to let the user know rather than proceeding with the query. Otherwise, jobs that leverage multiple threads may take much more resources. If runtime filtering stops working, jobs will simply start taking tons of time.